src/ContextualPopup/ContextualPopup.js

```diff
@@ -222,6 +229,18 @@ module.exports = kind(
 	initComponents: function () {
 		this.createChrome(this.tools);
 		Popup.prototype.initComponents.apply(this, arguments);
+		this.syncProperties();
+	},
+
+	/**
+	* Sync spotlightModal and modal properties on creation time.
+	*
+	* @private
+	*/
+	syncProperties: function () {
+		if (this.modal == this.spotlightModal) return;
+		if (this.hasOwnProperty('spotlightModal')) this.modal = this.spotlightModal;
```

Comment:
I don't have any actual cases for overriding both properties at creation time. So, simply sync like 'this.modal = this.spotlightModal' then add some documentation about its sync direction would be fine.